src/TurboMqtt/Client/IMqttClient.cs

```diff
@@ -290,7 +290,8 @@ public async Task<IConnectResponse> ConnectAsync(CancellationToken cancellationT
 
     public async Task DisconnectAsync(CancellationToken cancellationToken = default)
     {
-        if (_transport.Status != ConnectionStatus.Connected)
+        // already disconnected
+        if (_transport.Status is ConnectionStatus.Disconnected or ConnectionStatus.Aborted or ConnectionStatus.NotStarted)
```

Comment:
Bail out if we're not connected, but if we're connecting or in a failed state we can still try to gracefully shut down.